fix: rearrange bitwise operations while transform

wusyong opened the pull request:
While optimized transfomr function holds same results, operations in
each assignments are not the same number. So we reduce and rearrange
some operators back to original positions. Improved performance is still
valid.

Fix #120

marktwtn commented on src/pow_avx.c:
```diff
@@ -31,10 +31,9 @@ static void transform256(__m256i *lmid, __m256i *hmid)
-                                         one); /* ~(delta & alpha) */
+            delta = _mm256_and_si256( alpha, _mm256_xor_si256(lfrom[t2], beta));
+                    /* alpha & (lfrom[t2] ^ beta) */
+            lto[j] = _mm256_andnot_si256(delta, one); /* ~(delta & alpha) */
```
The comment needs to be revised to `~delta`.

marktwtn commented on src/pow_avx.c:
```diff
@@ -49,10 +48,9 @@ static void transform256(__m256i *lmid, __m256i *hmid)
-                                     one); /* ~(delta & alpha) */
+        delta = _mm256_and_si256( alpha, _mm256_xor_si256(lfrom[t2], beta));
+                /* alpha & (lfrom[t2] ^ beta) */
+        lto[j] = _mm256_andnot_si256(delta, one); /* ~(delta & alpha) */
```
Ditto.

marktwtn commented on src/pow_c.c:
```diff
@@ -43,9 +43,9 @@ static void transform64(uint64_t *lmid, uint64_t *hmid)
-        hto[j] = delta | (alpha ^ hfrom[t2]);
+        delta = alpha & (lfrom[t2] ^ beta);
+        lto[j] = ~delta;
+        hto[j] = (alpha ^ hfrom[t2]) | delta;
```
What is changing the position of `delta` for?
For consistency?

wusyong replied:
Yes, for consistency.

marktwtn commented:
> While optimized transfomr function holds same results, operations in
> each assignments are not the same number.

Typo: transfomr -> transform.

Commit:
fix: rearrange bitwise operations while transform

While optimized transform function holds same results, operations in
each assignments are not the same number. So we reduce and rearrange
some operators back to original positions. Improved performance is still
valid.

Fix #120
